PyLeap/Views/Project Views/ProjectViewModel.swift

```diff
@@ -38,4530 +35,351 @@ class ProjectViewModel: ObservableObject  {
+                  let name = resourceValues.name
+            else {
+                continue
             }
             
+            if isDirectory {
+                if name == "_extras" {
+                    dirEnumerator.skipDescendants()
+                }
+                projectDirectories.append(fileURL)
+            } else {
+                fileURLs.append(fileURL)
+            }
         }
         
-    }
-    
-    func ledGi2c_bit_Reg() {
-        
-        let documentsURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0].appendingPathComponent("RainbowBundle").appendingPathComponent("PyLeap_CPB_EyeLights_LED_Glasses_RainbowSwirl").appendingPathComponent("CircuitPython 7.x").appendingPathComponent("lib").appendingPathComponent("adafruit_register")
-        
-        guard let data = try? Data(contentsOf: URL(fileURLWithPath: "i2c_bit", relativeTo: documentsURL).appendingPathExtension("mpy")) else {
-            return
+        print("Listed Files")
+        for file in fileURLs {
+            print(file.lastPathComponent)
         }
-        
-        self.writeFileCommand(path: "/lib/adafruit_register/i2c_bit.mpy", data: data) { result in
-            switch result {
-            case .success:
-                self.ledGi2c_bits_Reg()
-                print("Success")
-            case .failure:
-                print("Failure - ledGi2c_bit_Reg")
-                
-            }
-            
+
+        DispatchQueue.main.async {
+            self.sendingBundle = true
         }
+        directoryCheck(dirList: projectDirectories, filesUrls: fileURLs)
     }
     
-    func ledGi2c_bits_Reg() {
-        
-        let documentsURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0].appendingPathComponent("RainbowBundle").appendingPathComponent("PyLeap_CPB_EyeLights_LED_Glasses_RainbowSwirl").appendingPathComponent("CircuitPython 7.x").appendingPathComponent("lib").appendingPathComponent("adafruit_register")
-        
-        guard let data = try? Data(contentsOf: URL(fileURLWithPath: "i2c_bits", relativeTo: documentsURL).appendingPathExtension("mpy")) else {
-            return
-        }
-        
-        
-        self.writeFileCommand(path: "/lib/adafruit_register/i2c_bits.mpy", data: data) { result in
-            switch result {
-            case .success:
-                self.ledGi2c_struct_array_Reg()
-                print("Success")
-            case .failure:
-                print("Failure - ledGi2c_bits_Reg")
+    func directoryCheck(dirList: [URL], filesUrls: [URL]) {
+        let tempDirectory = dirList
+
+        if dirList.isEmpty {
+            print("No directories left in queue")
+            self.transferFiles(files: filesUrls)
+        } else {
+            guard let directory = dirList.first else {
+                print("No directory exist here")
+                return
+            }
+
+            if directory.lastPathComponent == "lib" {
```

Comment:
Are you sure you need this? Could you just sort the paths so that shortest is first?